maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/extensions/E2ETest.java

```diff
@@ -184,4 +192,74 @@ public void close()
             .isPositive()
             .isLessThanOrEqualTo( 6_000L );
     }
+
+    @Test( timeout = 10_000L )
+    public void shouldAuthenticateClient() throws Exception
+    {
+        ForkNodeArguments forkNodeArguments = mock( ForkNodeArguments.class );
+        when( forkNodeArguments.getSessionId() ).thenReturn( UUID.randomUUID() );
+
+        try ( SurefireForkChannel server = new SurefireForkChannel( forkNodeArguments ) )
+        {
+            Thread t = new Thread()
+            {
+                @Override
+                public void run()
+                {
+                    SurefireMasterProcessChannelProcessorFactory client
+                        = new SurefireMasterProcessChannelProcessorFactory();
+                    try
+                    {
+                        client.connect( server.getForkNodeConnectionString() );
+                    }
+                    catch ( IOException e )
+                    {
+                        e.printStackTrace();
+                        throw new RuntimeException( e );
+                    }
+                }
+            };
+            t.setDaemon( true );
+            t.start();
+            server.connectToClient();
+        }
+    }
+
+    @Test( timeout = 10_000L )
+    public void shouldNotAuthenticateClient() throws Exception
+    {
+        ForkNodeArguments forkNodeArguments = mock( ForkNodeArguments.class );
+        when( forkNodeArguments.getSessionId() ).thenReturn( UUID.randomUUID() );
+
+        try ( SurefireForkChannel server = new SurefireForkChannel( forkNodeArguments ) )
+        {
+            Thread t = new Thread()
+            {
+                @Override
+                public void run()
+                {
+                    SurefireMasterProcessChannelProcessorFactory client
+                        = new SurefireMasterProcessChannelProcessorFactory();
+                    try
+                    {
+                        URI connectionString = new URI( server.getForkNodeConnectionString() );
+                        client.connect( "tcp://" + connectionString.getHost() + ":" + connectionString.getPort()
+                            + "?sessionId=6ba7b812-9dad-11d1-80b4-00c04fd430c8" );
+                    }
+                    catch ( Exception e )
+                    {
+                        e.printStackTrace();
```

Comment:
Same here

Reply:
See the [commit](https://github.com/apache/maven-surefire/pull/300/commits/c6f52fbb756482b0dfc3e5fae73c434b192db1fd) which explains that we did not expect an exception within the Thread. We expect it in server's facility verifying the session id returned by the client. Now the client's Thread also has to be running without exception.

Reply:
@michael-o This is no more printed. Handled with `Future.get()`.